ui/puzzle/src/keyboard.ts

```diff
@@ -33,9 +33,7 @@ export default (ctrl: KeyboardController) =>
     .bind('z', () => lichess.pubsub.emit('zen'))
     .bind('?', () => ctrl.keyboardHelp(!ctrl.keyboardHelp()))
     .bind('f', ctrl.flip)
-    .bind('n', () => {
-      if (ctrl.vm.mode === 'view') ctrl.nextPuzzle();
```

Comment:
Double checking that we don't need this conditional? IIRC, it was intended to disallow advancing the puzzle while you're still guessing.

Reply:
The check is now in nextPuzzle

Reply:
I moved this check to https://github.com/lichess-org/lila/pull/11354/files#diff-ed9feaa882736ee2bff4d3a3dacbb32078212ed1ae48fc013edf92f1093f2994R348 so it's performed both for `n` and `next` keyboard commands.

Reply:
Nice, definitely better there